src/core/model.jl

```diff
@@ -5,24 +5,26 @@ export ABM, AgentBasedModel
 #######################################################################################
 
 abstract type AbstractSpace end
-SpaceType=Union{Nothing, AbstractSpace}
+SpaceType = Union{Nothing,AbstractSpace}
 
 abstract type DiscreteSpace <: AbstractSpace end
 
 # This is a collection of valid position types, sometimes used for ambiguity resolution
-ValidPos = Union{Int, NTuple{N, Int}, NTuple{M, <:AbstractFloat}, Tuple{Int, Int, Float64}} where {N, M}
+ValidPos =
+    Union{Int,NTuple{N,Int},NTuple{M,<:AbstractFloat},Tuple{Int,Int,Float64}} where {N,M}
 
-struct AgentBasedModel{S<:SpaceType, A<:AbstractAgent, F, P}
+struct AgentBasedModel{S<:SpaceType,A<:AbstractAgent,F,P}
     agents::Dict{Int,A}
     space::S
     scheduler::F
     properties::P
+    rng::AbstractRNG
```

Comment:
`AbstractRNG` is an abstract type (says so in the name)! You need to parameterize this like the other fields.